asp_net-IVR/README.md

```diff
@@ -0,0 +1,103 @@
+|:-----------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `TELNYX_API_KEY`       | Your [Telnyx API Key](https://portal.telnyx.com/#/app/api-keys?utm_source=referral&utm_medium=github_referral&utm_campaign=cross-site-link)              |
+| `TELNYX_PUBLIC_KEY`    | Your [Telnyx Public Key](https://portal.telnyx.com/#/app/account/public-key?utm_source=referral&utm_medium=github_referral&utm_campaign=cross-site-link) |
+| `PORT`      | **Defaults to `8000`** The port the app will be served                                                                                                   |
+
+### .env file
+
+This app uses the excellent [dotenv.net](https://github.com/bolorundurowb/dotenv.net) package to manage environment variables.
+
+Make a copy of [`.env.sample`](./.env.sample) and save as `.env` and update the variables to match your creds.
+
+```
+TELNYX_API_KEY=
+TELNYX_PUBLIC_KEY=
+PORT=8000
+```
+
+### Callback URLs For Telnyx Applications
+
+| Callback Type                    | URL                              |
+|:---------------------------------|:---------------------------------|
+| Inbound Calls Callback         | `{ngrok-url}/call-control/inbound`  |
+| Outbound Calls Callback | `{ngrok-url}/call-control/outbound` |
+
+### Install
+
+Run the following commands to get started
+
+```
+$ git clone https://github.com/team-telnyx/demo-dotnet-telnyx.git
+```
+
+### Ngrok
+
+This application is served on the port defined in the runtime environment (or in the `.env` file). Be sure to launch [ngrok](https://developers.telnyx.com/docs/v2/development/ngrok?utm_source=referral&utm_medium=github_referral&utm_campaign=cross-site-link) for that port
+
+```
+./ngrok http 8000
+```
+
+> Terminal should look _something_ like
+
+```
+ngrok by @inconshreveable                                                                                                                               (Ctrl+C to quit)
+
+Session Status                online
+Account                       Little Bobby Tables (Plan: Free)
+Version                       2.3.35
+Region                        United States (us)
+Web Interface                 http://127.0.0.1:4040
+Forwarding                    http://your-url.ngrok.io -> http://localhost:8000
+Forwarding                    https://your-url.ngrok.io -> http://localhost:8000
+
+Connections                   ttl     opn     rt1     rt5     p50     p90
+                              0       0       0.00    0.00    0.00    0.00
+```
+
+At this point you can point your application to generated ngrok URL + path  (Example: `http://{your-url}.ngrok.io/call-control/inbound`).
+
+### Run
+
+Open your IDE and run the application and call the phone number associated with your call control application.
+
+You'll be prompted to enter a 10 digit phone number then the app will transfer to the number entered.
```

Comment:
Should I be attempting to run this in Visual Studio or IntelliJ IDEA?

Gave it a go with just `dotnet`:

```
drew@Drews-MacBook-Pro asp_net-IVR % dotnet run
Building...
info: Microsoft.Hosting.Lifetime[0]
      Now listening on: http://[::]:80
info: Microsoft.Hosting.Lifetime[0]
      Now listening on: https://localhost:8001
info: Microsoft.Hosting.Lifetime[0]
      Application started. Press Ctrl+C to shut down.
info: Microsoft.Hosting.Lifetime[0]
      Hosting environment: Development
info: Microsoft.Hosting.Lifetime[0]
      Content root path: /Users/drew/repos/team-telnyx/demo-dotnet-telnyx/asp_net-IVR
```

Unclear how I should proceed to trigger prompt for phone number.

Reply:
added clarity, in this case it's triggered by an inbound call.